hw2/02-search.js

```diff
@@ -1,3 +1,65 @@
-// Add your code here
+import { characters } from "./02-data.js";
 
-userInput.addEventListener('click', handleClick);
+const userInput = document.getElementById("userInput");
+const searchButton = document.getElementById("search");
+const cardContainer = document.getElementById("cardContainer");
+
+function createCard(title, body) {
+  const card = document.createElement("div");
+  const cardContent = document.createElement("div");
+  const cardTitle = document.createElement("h5");
+  const cardBody = document.createElement("p");
+  card.className = "card justify-content-center";
+  cardContent.className = "card-body";
+  card.style.maxWidth = "200px";
+  cardTitle.innerHTML = title;
+  cardBody.textContent = body;
+  cardContent.appendChild(cardTitle);
+  cardContent.appendChild(cardBody);
+  card.appendChild(cardContent);
+  cardContainer.appendChild(card);
+  return card;
+}
+
+function handleClick() {
+  const characterQuery = userInput.value;
+  const matchingCharacters = [];
+
+  //   Clear the container before performing a new search
+  cardContainer.innerHTML = "";
+
+  if (characterQuery === "") {
+    const card = createCard(
+      "Didn't provide Anything",
+      "Please try typing a charcter's name :)",
+    );
+    card.style.color = "red";
+    return;
+  }
+
+  characters.forEach((character) => {
+    if (character.name.includes(characterQuery)) {
```

Comment:
Hey Kobe, jut a suggestion here, I think we can cover edge cases as well if we check this condition with equalsIgnoreCase() :)